src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3233,18 +3234,38 @@ static size_t large_page_init_decide_size() {
     return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  if (!os::win32::is_windows_11_or_greater() && !os::win32::is_windows_server_2022_or_greater()) {
+#if defined(IA32)
+    if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+      WARN("JVM cannot use large pages bigger than 4mb.");
+      return 0;
+    }
+#elif defined(AMD64)
+    if (!EnableAllLargePageSizes) {
+      if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+        WARN("JVM cannot use large pages bigger than 4mb.");
+        return 0;
+      }
+    } else {
+      WARN("EnableAllLargePageSizes flag is ignored on Windows versions prior to Windows 11/Windows Server 2022 due to limited support.");
+    }
 #endif
+  }
 
-  if (LargePageSizeInBytes > 0 && LargePageSizeInBytes % size == 0) {
-    size = LargePageSizeInBytes;
+  if (LargePageSizeInBytes > 0) {
+    if (LargePageSizeInBytes % size == 0) {
+      size = LargePageSizeInBytes;
+    }
+    else {
+      WARN1("The specified large page size (%d) is not a multiple of the minimum large page size (%d), defaulting to minimum page size.", LargePageSizeInBytes, size);
+    }
```

Comment:
NIT: you can make these two "if's" into one and use just a "?" operator to print different messages.

Reply:
Do you think I should introduce more warn macros , as warn macros accepts only one argument or the propsed change where I am using a buffer would be a better solution

Reply:
> NIT: you can make these two "if's" into one and use just a "?" operator to print different messages.

Do you think I should introduce more warn macros , as warn macros accepts only one argument or the propsed change where I am using a buffer would be a better solution

Reply:
It's up to you man. Don't overthink it, it's well possible other reviewers may think differently of us.

Reply:
sure, then in that case, let me test with this and I will resolve the conversation.